app/models/employment/AllEmploymentData.scala

```diff
@@ -37,7 +40,18 @@ case class EmploymentSource(employmentId: String,
                             dateIgnored: Option[String],
                             submittedOn: Option[String],
                             employmentData: Option[EmploymentData],
-                            employmentBenefits: Option[EmploymentBenefits]){
+                            employmentBenefits: Option[EmploymentBenefits]) extends Logging {
+
+  def getSubmittedOnDateTime: Option[ZonedDateTime] ={
```

Comment:
I think this logic can be pulled out into some DateTimeParser object since its used in a few places?

Also you could get rid of the "Java-ry" try catch and use scalas Try